CMakeLists.txt

```diff
@@ -133,6 +133,7 @@ if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
 endif()
 if((CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME) AND BUILD_TESTING)
     add_subdirectory(tests)
+    add_subdirectory(bench)
```

Comment:
Do we want to always run the benchmarks when testing is enabled?

The benchmarks require third-parties, and internet connection. Having the possibility to disable them would be beneficial, am I right?

Reply:
Considering the need for a network connection I do agree. I've added a cmake variable to control whether the benchmarks get built, and have them turned them off by default.